test/test_util.py

```diff
@@ -360,6 +360,20 @@ def test_request_uri(self, url: str, expected_request_uri: str) -> None:
         returned_url = parse_url(url)
         assert returned_url.request_uri == expected_request_uri
 
+    url_authority_map = [
+        ("http://user:pass@google.com/mail", "user:pass@google.com"),
```

Comment:
Whenever I'm testing URLs I try to hit the follow variants:
- single label hostname (`localhost`)
- multi label hostname
- ipv4
- ipv6

And then all of the above with and without a port.

Reply:
Since this is supposed to be the same as urlparse().netloc can we also verify that each test case that starts with `http://` matches between these two APIs?

Reply:
I left the constraint a little looser looking for `startswith('http')` instead of fully qualified `http://`. My goal is to be greedier with new test cases with things like `https` and `http+`. That can fail at PR time if that match is wrong, and we'll address it then, rather that think we added a test case but it's silently skipped.

Happy to change if there's consensus we'd rather tighten the scope.